R/TileDBArray.R

```diff
@@ -7,13 +7,32 @@ TileDBArray <- R6::R6Class(
   public = list(
     uri = NULL,
     verbose = TRUE,
+    #' @field optional configuration'
+    config = NULL,
+    #' @field optional tiledb context'
+    ctx = NULL,
 
     #' @description Create a new TileDBArray object.
     #' @param uri URI for the TileDB array
     #' @param verbose Print status messages
-    initialize = function(uri, verbose = TRUE) {
+    #' @param config optional configuration
+    #' @param ctx optional tiledb context
+    initialize = function(uri, verbose = TRUE, config = NULL, ctx = NULL) {
       self$uri <- uri
       self$verbose <- verbose
+      self$config <- config
+      self$ctx <- ctx
```

Comment:
```suggestion
      self$ctx <- ctx %||% tiledb::tiledb_get_context()
```

Could use our infix operator to replace NULLs with a default and remove the lines 32-34.